src/ParallelAlgorithms/Interpolation/Actions/InterpolatorReceiveVolumeData.hpp

```diff
@@ -48,6 +48,41 @@ struct InterpolatorReceiveVolumeData {
       const typename TemporalId::type& temporal_id,
       const ElementId<VolumeDim>& element_id, const ::Mesh<VolumeDim>& mesh,
       Variables<typename Metavariables::interpolator_source_vars>&& vars) {
+    // Determine if we have already finished interpolating on this
+    // temporal_id.  If so, then we simply return, ignore the incoming
+    // data, and do not interpolate.
+    //
+    // This scenario can happen if there is an element that is not
+    // used or needed for any InterpolationTarget, and if that element
+    // calls InterpolatorReceiveVolumeData so late that all the
+    // InterpolationTargets for the current temporal_id have already
+    // finished.
+    bool this_temporal_id_is_done = true;
+    const auto& holders =
+        db::get<Tags::InterpolatedVarsHolders<Metavariables>>(box);
+    tmpl::for_each<typename Metavariables::interpolation_target_tags>(
+        [&](auto tag_v) {
+          using tag = typename decltype(tag_v)::type;
+          if constexpr (std::is_same_v<TemporalId, typename tag::temporal_id>) {
+            const auto& finished_temporal_ids =
+                get<Vars::HolderTag<tag, Metavariables>>(holders)
+                    .temporal_ids_when_data_has_been_interpolated;
+            if (finished_temporal_ids.count(temporal_id) == 0) {
```

Comment:
[optiona] shorten to `this_temporal_id_is_done = finished_temporal_ids.count(temporal_id) != 0;`

Reply:
I don't think that is correct.  `this_temporal_id_is_done` should be true only if *all* of the `interpolation_target_tags` are done, so we never want to set `this_temporal_id_is_done` to true inside the loop (because 'true' might occur for the last loop iteration).

Reply:
ah, okay, yes, sorry!